From eb34ea45fe41a468b02236cc2022c9143bbced8f Mon Sep 17 00:00:00 2001
From: Brian Delwiche <delwiche@google.com>
Date: Thu, 31 Oct 2024 00:05:16 +0000
Subject: [PATCH] Fix incorrect logging in sdp_server

In the function process_service_search_attr_req of sdp_server.cc, a log
statement uses a structure that may, in exceptional cases, have been
freed by preceding calls.  This can lead to use after free and
potentially to security vulnerabilities.

Use local variables instead.

Bug: 375407167
Test: m libbluetooth
Test: researcher POC
Flag: EXEMPT trivial logic fix
Ignore-AOSP-First: security
Tag: #security
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:f23300552c43a4f503debcf0236f29211ee1c557)
Merged-In: I1037e592ab8910eeae5c28a41a53a2fc60a298d8
Change-Id: I1037e592ab8910eeae5c28a41a53a2fc60a298d8
---
 system/stack/sdp/sdp_server.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/system/stack/sdp/sdp_server.cc b/system/stack/sdp/sdp_server.cc
index 7061cc5885..7566e7e3d2 100644
--- a/system/stack/sdp/sdp_server.cc
+++ b/system/stack/sdp/sdp_server.cc
@@ -564,7 +564,7 @@ static void process_service_attr_req(tCONN_CB* p_ccb, uint16_t trans_num, uint16
   if (stack::l2cap::get_interface().L2CA_DataWrite(p_ccb->connection_id, p_buf) !=
       tL2CAP_DW_RESULT::SUCCESS) {
     log::warn("Unable to write L2CAP data peer:{} cid:{} len:{}", p_ccb->device_address,
-              p_ccb->connection_id, p_buf->len);
+              p_ccb->connection_id, p_rsp - p_rsp_start);
   }
 }
 
-- 
2.34.1

